### PR TITLE
Task/adjust select styles

### PR DESCRIPTION
**Pull request by gomezjuliana**
Hello! Here are the updates to the Selects :)

Figma file: https://www.figma.com/file/Of9hvMKpQQxG0DusWCJPejcp/BDS-Select?node-id=0%3A1

Happy to make any changes needed!

### CODE REVIEW

**Comment by twanlass** [comment]
Heya @gomezjuliana – is there a dev / staging environment where I can preview these changes?

**Comment by twanlass** [comment]
Heya @gomezjuliana – sorry, should have pulled and built this locally 🙈. 

**Just a few odds and ends left:**
- Let's remove the triangle / caret 
- Margin / padding missing on the right and left of the menu item
- 10px of extra left padding applied to text only menu items
- Checkmark icon should be same color as menu item text `#3D3D3D`

**Reference:**
select-edits


**Quick Figma tip:**
You can use `ctrl+click` to select a specific element. Then holding `alt` and mousing over other elements you can see the margins and other dimensions:
figma-dimensions

**Comment by gomezjuliana** [comment]
Thanks so much for the feedback @twanlass 😁 Here are the changes! We only have local + prod envs right now but I'll add it to the agenda on Tuesday to see if it's something we could set up for the future.
Let me know if I've missed anything 😁

**Comment by twanlass** [comment]
Heya @gomezjuliana – this is looking so good. Yaasss! 

**I have two small interaction notes / details:** 
- For multi-selects, can we always preserve the left margin (placeholder margin for the checkmark) so that the text doesn't shift when items are selected / unselected? Same would be true for a select w/ search. [[ref](http://hi.buffer.com/c09a3b4959f5)]
- It looks like we lose the selected state while searching (note that the checkmark disappears while searching). [[ref](http://hi.buffer.com/74d9a062dedd)]

**Comment by gomezjuliana** [comment]
Heya @ivanaszuber,

I added in the style changes for Tyler's first point but I wonder if you could help me with the second and third?

For the second, I noticed that when we select an option we're adding `selected` to the state but in `onSearchChange` we're filtering from props so that could be why the checkmark isn't sticking? I also noticed that when we `handleSelectOption` we might not be updating the deselected/selected items.. I tried to look into this but I couldn't find the docs to `helper` from immutibility-helper 😅 

Happy to chat through this tomorrow if you'd like!

**Comment by ivanaszuber** [comment]
Heya @gomezjuliana , yeah, that's a really great one! I took a look at the code, and indeed, we'll need to make a few changes in order to have this working properly.
I'll push a few changes to this branch, but let's also jump on a quick sync to discuss those today if you're up for it 😊 🙌

File: src/components/Select/Select.jsx [hunk 1]
```diff
@@ -22,12 +22,17 @@ export default class Select extends React.Component {
...
+        el &&
+        itemsB[ix] &&
+        el.id === itemsB[ix].id &&
+        el.title === itemsB[ix].title
     );
 
   state = {
     isOpen: this.props.isOpen,
     items: this.props.items || [],
+    selectedItems: this.props.items || [],
```
Comment (ivanaszuber) [inline src/components/Select/Select.jsx]: we're adding another items array here to the state in order to use it later when filtering/searching the items

File: src/components/Select/Select.jsx [hunk 2]
```diff
@@ -103,6 +108,15 @@ export default class Select extends React.Component {
     }
   };
 
+  updateItemsInState = (items, option, index) =>
+    index > -1
+      ? helper(items, {
+          [index]: {
+            selected: { $set: !option.selected },
```
Comment (ivanaszuber) [inline src/components/Select/Select.jsx]: update the `selected` prop in items

File: src/components/Select/Select.jsx [hunk 3]
```diff
@@ -119,18 +133,20 @@ export default class Select extends React.Component {
...
-              [optionIndex]: {
-                selected: { $set: !option.selected },
-              },
-            })
-          : items,
+      items: this.updateItemsInState(deselectItems, option, optionIndex),
+      // we need to copy the items to another array here in order to use that one during search
+      // filtering
+      selectedItems: this.updateItemsInState(
+        deselectItems,
```
Comment (ivanaszuber) [inline src/components/Select/Select.jsx]: whenever we're selecting/deselecting an item, update the `selectedItems` as well in order to be able to use it later when searching/filtering

File: src/components/Select/Select.jsx [hunk 4]
```diff
@@ -252,13 +268,28 @@ export default class Select extends React.Component {
     }
   };
 
+  findItemInState = item => {
+    const { selectedItems } = this.state;
+    return selectedItems.find(x => this.getItemId(x) === this.getItemId(item));
```
Comment (ivanaszuber) [inline src/components/Select/Select.jsx]: Find the given item in the `selectedItems` list

File: src/components/Select/Select.jsx [hunk 5]
```diff
@@ -252,13 +268,28 @@ export default class Select extends React.Component {
...
+    const filteredItems = items.reduce((filtered, item) => {
+      if (
+        includes(item[searchFiled].toLowerCase(), searchValue.toLowerCase())
+      ) {
+        filtered.push({
+          ...item,
+          selected:
+            this.findItemInState(item) && this.findItemInState(item).selected,
+        });
+      }
```
Comment (ivanaszuber) [inline src/components/Select/Select.jsx]: For each item in `props.items` , if the item contains the search string, add that item to the `filtered` list and set the `selected` state of that item to be the one from the `selectedItems`

**Pull request merged**